README.md

```diff
@@ -66,18 +66,83 @@ If you need to set a default configuration
 $ spack install celeritas +cuda cuda_arch=80
 ```
 
-
-Then see the "Downstream usage as a library" section of the [installation
-documentation][install] for how to use Celeritas in your application or framework.
-
 [spack-start]: https://spack.readthedocs.io/en/latest/getting_started.html
 [install]: https://celeritas-project.github.io/celeritas/user/usage/installation.html
 
+# Integrating into a Geant4 app
+
+In the simplest case, integration requires a small change to your project's
+CMake file:
+```diff
++find_package(Celeritas 0.6 REQUIRED)
+ find_package(Geant4 REQUIRED)
+@@ -36,3 +37,4 @@ else()
+   add_executable(trackingmanager-offload trackingmanager-offload.cc)
+-  target_link_libraries(trackingmanager-offload
++  celeritas_target_link_libraries(trackingmanager-offload
++    Celeritas::accel
+     ${Geant4_LIBRARIES}
+```
+
+a few includes:
+```diff
+--- example/accel/trackingmanager-offload.cc
++++ example/accel/trackingmanager-offload.cc
+@@ -36,2 +36,10 @@
+
++// Celeritas
++#include <accel/AlongStepFactory.hh>
++#include <accel/SetupOptions.hh>
++#include <accel/TrackingManagerConstructor.hh>
++#include <accel/TrackingManagerIntegration.hh>
++
++using TMI = celeritas::TrackingManagerIntegration;
++
+ namespace
+```
+
+minor additions to the user actions:
+```diff
+--- example/accel/trackingmanager-offload.cc
++++ example/accel/trackingmanager-offload.cc
+@@ -134,2 +142,3 @@ class RunAction final : public G4UserRunAction
+     void BeginOfRunAction(G4Run const* run) final {
++        TMI::Instance().BeginOfRunAction(run);
+     }
+@@ -137,2 +146,3 @@ class RunAction final : public G4UserRunAction
+     void EndOfRunAction(G4Run const* run) final {
++        TMI::Instance().EndOfRunAction(run);
+     }
+@@ -179,2 +189,3 @@ class ActionInitialization final : public G4VUserActionInitialization
+      void BuildForMaster() const final {
++        TMI::Instance().BuildForMaster();
+     }
+@@ -185,2 +197,3 @@ class ActionInitialization final : public G4VUserActionInitialization
+     void Build() const final{
++        TMI::Instance().Build();
+     }
+```
+and addition to the physics list:
+```diff
+--- example/accel/trackingmanager-offload.cc
++++ example/accel/trackingmanager-offload.cc
+@@ -203,4 +235,8 @@ int main()
+
++    auto& tmi = TMI::Instance();
++
+     // Use FTFP_BERT, but use Celeritas tracking for e-/e+/g
+     auto* physics_list = new FTFP_BERT{/* verbosity = */ 0};
++    physics_list->RegisterPhysics(
++        new celeritas::TrackingManagerConstructor(&tmi));
```

Comment:
Side question: since ADePT says that this is the only part needed for ADePT what are the additional capabilities instructmenting the Run and User action brings to Celeritas?

Reply:
Run action allows us to wait until the physics, geometry, etc. are actually set up before use, and unlike adept we actually clean up after ourselves. The `ActionInitialization` lets us set up the logging before anything else but it may not be strictly necessary... maybe we should delete this in 0.7.

Reply:
Maybe adding a "Celeritas.hh" would be good to reduce the includes too?

Reply:
I'll do this extra simplification as a follow-on PR.